desktop/dev-environments.md

```diff
@@ -253,6 +253,21 @@ In this preview, Dev Environments support a JSON file which allows you to specif
 }
 ```
 
+Next, you have to define the dependencies you want to include in your `Dockerfile.devenv`, alongside the following requisites:
+
+While any images or Dockerfiles will include a non-root user with a UID/GID of `1000`, many base images and Dockerfiles do not. Fortunately, you can add a non-root user named `vscode`. If you were to include the Docker tooling (e.g. `docker` cli, `docker compose`, etc.) in the `Dockerfile.devenv`, you would need the `vscode` user to be included in the `docker` group. Finally, remember to set `sleep infinity` in the entrypoint to keep the dev container running indefinitely.
+
+```dockerfile
+FROM <your base image>
+
+RUN useradd -s /bin/bash -m vscode
+RUN groupadd docker && usermod -aG docker vscode
```

Comment:
This will create 2 layers; perhaps we should combine these;

```suggestion
RUN useradd -s /bin/bash -m vscode \
 && groupadd docker \
 && usermod -aG docker vscode
```

Do we require `/bin/bash` or does `/bin/sh` work as well? `/bin/bash` won't be available by default in many images.

These should likely use `useradd` instead of `adduser`, because `adduser` is an optional install (`useradd` / `groupadd` are the default; see https://superuser.com/a/1543631)

When updating, also be sure to test them on `rpm` (e.g. `fedora` or `centos`), `apk` (alpine), and `deb` (`ubuntu` or `debian`) based base images; unfortunately the command flags differ a bit between distros 😞 (see the Moby code, which also has some variants for this); https://github.com/moby/moby/blob/7b9275c0da707b030e62c96b679a976f31f929d3/pkg/idtools/usergroupadd_linux.go#L65-L88

The Dockerfile for the nginx image may be useful to find the corresponding variants for alpine and debian;

- alpine: https://github.com/nginxinc/docker-nginx/blob/01561128671d4e33c039a654fe198c6259a2514c/mainline/alpine/Dockerfile#L16-L17
- debian: https://github.com/nginxinc/docker-nginx/blob/3a7105159a6c743188cb1c61e4186a9a59c025db/mainline/debian/Dockerfile#L16-L17